File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i, j = 0, 1
        profit = 0
        while j < len(prices):
            if prices[j] < prices[i]:
                i = j
            profit = max(profit, prices[j] - prices[i])
            j += 1
        return profit

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    largestRectangleArea(heights) {
        const n = heights.length
        let st = []
        let res = 0

        heights.push(0)

        for(let i = 0; i < n + 1; i++) {
            while(st.length > 0 && heights[i] < heights[st[st.length - 1]]) {
                const h = heights[st.pop()]
                const w = st.length > 0 ? i - st[st.length - 1] - 1 : i

                res = Math.max(res, h*w)
            } 
            st.push(i)
        }

        return res
    }
}
